cmd/realm/triggers.go

```diff
@@ -69,111 +71,357 @@ var triggersDeleteCmd = &cobra.Command{
+		os.Exit(1)
+	}
 
-	getTriggerCall, err := astarteAPIClient.GetTrigger(realm, triggerName)
+	var targetPath string
+	var err error
+	if len(args) == 0 {
+		targetPath, _ = filepath.Abs(".")
+	} else {
+		targetPath, err = filepath.Abs(args[0])
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+	}
+
+	// retrieve triggers list
+	realmTriggers, err := listTriggers(realm)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 
-	utils.MaybeCurlAndExit(getTriggerCall, astarteAPIClient)
+	for _, name := range realmTriggers {
 
-	getTriggerRes, err := getTriggerCall.Run(astarteAPIClient)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
+		triggerDefinition, err := getTriggerDefinition(realm, name)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		respJSON, err := json.MarshalIndent(*triggerDefinition, "", "  ")
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		filename := fmt.Sprintf("/%s/%s.json", targetPath, name)
+		outFile, err := os.Create(filename)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		defer outFile.Close()
+
+		if _, err := outFile.Write(respJSON); err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+	}
+	return nil
+}
+
+func triggersSyncF(command *cobra.Command, args []string) error {
+	if viper.GetBool("to-curl") {
+		fmt.Println(`'triggers sync' does not support the --to-curl option. Install your triggers one by one with 'triggers install'.`)
 		os.Exit(1)
 	}
 
-	triggerDefinition, _ := getTriggerRes.Parse()
-	respJSON, _ := json.MarshalIndent(triggerDefinition, "", "  ")
-	fmt.Println(string(respJSON))
+	triggersToInstall := []triggers.AstarteTrigger{}
+	triggersToUpdate := []triggers.AstarteTrigger{}
+	invalidTriggers := []string{}
+
+	for _, f := range args {
+		triggerFile, err := os.ReadFile(f)
+		if err != nil {
+			return err
+		}
+		if !validateTrigger(f) {
+			invalidTriggers = append(invalidTriggers, f)
+			continue
+		}
+
+		var astarteTrigger triggers.AstarteTrigger
+		if err = json.Unmarshal(triggerFile, &astarteTrigger); err != nil {
+			return err
+		}
+
+		if _, err := getTriggerDefinition(realm, astarteTrigger.Name); err != nil {
+			// The trigger does not exist
+			triggersToInstall = append(triggersToInstall, astarteTrigger)
+		} else {
+			triggersToUpdate = append(triggersToUpdate, astarteTrigger)
+		}
+
+		if len(triggersToInstall) == 0 && len(triggersToUpdate) == 0 {
+			// All good in the hood
+			fmt.Println("Your realm is in sync with the provided triggers files")
+			return nil
+		}
+
+	}
+	// Notify the user about what we're about to do
+	list := []string{}
+
+	for _, trigger := range triggersToInstall {
+		list = append(list, trigger.Name)
+	}
+
+	list_existing := []string{}
```

Comment:
Let's keep naming consistency
```suggestion
	listExisting := []string{}
```